src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderStrips.java

```diff
@@ -237,27 +240,51 @@ private void interpretStrip(
 
     @Override
     public void readImageData(final ImageBuilder imageBuilder)
-            throws ImageReadException, IOException {
-        for (int strip = 0; strip < imageData.getImageDataLength(); strip++) {
-            final long rowsPerStripLong = 0xFFFFffffL & rowsPerStrip;
-            final long rowsRemaining = height - (strip * rowsPerStripLong);
-            final long rowsInThisStrip = Math.min(rowsRemaining, rowsPerStripLong);
-            final long bytesPerRow = (bitsPerPixel * width + 7) / 8;
-            final long bytesPerStrip = rowsInThisStrip * bytesPerRow;
-            final long pixelsPerStrip = rowsInThisStrip * width;
-
-            final byte[] compressed = imageData.getImageData(strip).getData();
-
-            final byte[] decompressed = decompress(compressed, compression,
-                    (int) bytesPerStrip, width, (int) rowsInThisStrip);
-
-            interpretStrip(
-                    imageBuilder,
-                    decompressed,
-                    (int) pixelsPerStrip,
-                    height);
-
+      throws ImageReadException, IOException {
+        if (planarConfiguration != 2) {
```

Comment:
Is `1` the equivalent of `TiffTagConstants.PLANAR_CONFIGURATION_VALUE_PLANAR`? If so, I think we should use the constant?

Reply:
You are absolutely right about using the TiffTagConstant values in the places you cited.  Actually, I wanted to make line 244 test for planarConfiguration == "Chunky" rather than != "Planar", but I was afraid that there might be places in the code where the planarConfiguration variable didn't get set correctly.  So I thought this would be safer.

If we ever get our hands on some additional test files, I might revisit this.  But that may be unlikely...  The Tiff Specification itself describes planar-configuration 2 as follows: _PlanarConfiguration=2 is not currently in widespread use and it is not recommended for general interchange. It is used an extension and Baseline TIFF readers are not required to support it._